src/gsi/read_prepbufr.f90

```diff
@@ -2003,6 +2003,85 @@ subroutine read_prepbufr(nread,ndata,nodata,infile,obstype,lunout,twindin,sis,&
 ! don't use MESONET psfc obs if  8th character of station id is "x")
               if( kx==188 .and. psob .and. sidchr(8)=='x' ) usage=r100
 
+!             Set inflate_error logical based on qm flag
```

Comment:
@jderber-NOAA  Would you let me know what purpose is for this part of changes. Maye, it is only some simple moving of blocks for uvob. But it is difficult for me to trace the changes.  will these changes be supposed to change the results ?  The current regession tests are enough to verify changes here? (the GSI would go through this part)

Reply:
@jderber-NOAA I think if changes in read_prepbufr.f90 could be put in another PR, the changes for processing VAD could be rather quickly add to the GSI repository through this PR.

Reply:
The change here is the important change to resolve the issue.  The processing of the VAD winds needs the surface elevation stuff so it has to be moved along with the actual VAD wind stuff.  As the code was, the VAD processing occurred after iout was determined.  Since the VAD wind processing jumps out of the loop sometimes, when it jumps out of the k loop it results in the cdata_ array being all zero for that value of iout.

Reply:
By the way, the results will not change.

Reply:
Thanks for this explanation!